ext/mathematical/mathematical.c

```diff
@@ -137,15 +128,13 @@ static VALUE MATHEMATICAL_process(VALUE self, VALUE rb_LatexCode, VALUE rb_TempF
   g_object_unref (view);
   g_object_unref (document);
 
-  char* svg_contents = readFile(tempfile);
-
-  if (svg_contents == NULL) rb_raise(rb_eDocumentReadError, "Failed to read SVG contents");
+  if (rb_iv_get(self, "@svg") == Qnil) rb_raise(rb_eDocumentReadError, "Failed to read SVG contents");
 
   VALUE result_hash = rb_hash_new();
 
   rb_hash_aset (result_hash, rb_tainted_str_new2 ("width"),  INT2FIX(width_pt));
   rb_hash_aset (result_hash, rb_tainted_str_new2 ("height"), INT2FIX(height_pt));
-  rb_hash_aset (result_hash, rb_tainted_str_new2 ("svg"),    rb_str_new2(svg_contents));
+  rb_hash_aset (result_hash, rb_tainted_str_new2 ("svg"),    rb_iv_get(self, "@svg"));
```

Comment:
No ofcour  not  but when you did go the first time to the you rite you name from the first day